management-interface.go

```diff
@@ -289,6 +293,301 @@ func NetworkHandler(w http.ResponseWriter, r *http.Request) {
 	tmpl.ExecuteTemplate(w, "network.html", state)
 }
 
+//getNetworkSsid -gets the ssid of the network associated with this id
+func getNetworkSsid(networkID string) (string, error) {
+	var out []byte
+	err := error(nil)
+	//get network
+	out, err = exec.Command("wpa_cli", "get_network", networkID, "ssid").Output()
+	if err != nil {
+		log.Printf("error executing wpa_cli get_network %s - error %s output %s", networkID, err.Error(), out)
+		return "", err
+	}
+
+	stdOut := string(out)
+	scanner := bufio.NewScanner(strings.NewReader(stdOut))
+	scanner.Scan() //skip 1st line interface line
+	scanner.Scan()
+	ssid := scanner.Text()
+	return ssid, err
+
+}
+
+//deleteNetwork - delets network with id using wpa_cli and saves config
+func deleteNetwork(id string) error {
+	var out []byte
+	err := error(nil)
+
+	//check if is bushnet
+	var ssid string
+	ssid, err = getNetworkSsid(id)
+	if strings.ToLower(ssid) == "\"bushnet\"" {
+		log.Printf("error bushnet cannot be deleted")
+		return err
+	}
+
+	//remove network
+	cmd := exec.Command("wpa_cli")
+	var stdin io.WriteCloser
+	stdin, err = cmd.StdinPipe()
+	if err != nil {
+		log.Printf("error getting stdin pipe from cmd -error %s", err.Error())
+		return err
+	}
+	defer stdin.Close()
+	io.WriteString(stdin, fmt.Sprintf("remove_network %s\n", id))
+	io.WriteString(stdin, "quit\n")
+	out, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("error deleting wpa network -error %s", err.Error())
+		return err
+	}
+	errOccured := hasErrorOccured(string(out))
+	if errOccured {
+		reloadWpaConfig()
+		err = errors.New("error deleting network")
+		log.Printf(err.Error())
+		return err
+	}
+
+	//save and reload config
+	err = saveWpaConfig()
+	reloadErr := reloadWpaConfig()
+	if err == nil { //probably wont happen
+		err = reloadErr
+	}
+	return err
+}
+
+//doesWpaNetworkExist - checks if ssid is already specified in wpa_supplicant
+func doesWpaNetworkExist(ssid string) bool {
+	exists := false
+	networks, _ := parseWpaSupplicantConfig()
```

Comment:
if there's an error better log it and return false

Reply:
is it best to log it and return bool and error ?

Reply:
Returning `(bool, error)` and surfacing the error to the user would be best.

Either log it and return bool, or *don't* log it and return the error. Both logging and returning an error is almost always the wrong thing to do :)